src/screens/NFTSingleOfferSheet/index.tsx

```diff
@@ -136,16 +210,228 @@ export function NFTSingleOfferSheet() {
       return () => clearInterval(interval);
     }
   }, [offer.validUntil]);
-  const isExpiring =
-    timeRemaining !== undefined && timeRemaining <= TWO_HOURS_MS;
-  const isExpired = timeRemaining === 0;
-  const time = timeRemaining
-    ? getFormattedTimeQuantity(timeRemaining)
-    : undefined;
-  const buttonColorFallback = useForegroundColor('accent');
 
-  const feesPercentage = Math.floor(offer.feesPercentage * 10) / 10;
-  const royaltiesPercentage = Math.floor(offer.royaltiesPercentage * 10) / 10;
+  // estimate gas
+  useEffect(() => {
+    if (!isReadOnlyWallet && !isExpired) {
+      (async () => {
+        const privateKey = await loadPrivateKey(accountAddress, false);
```

Comment:
this is going to trigger faceId on mount, we should be able to pass in a read only signer for gas estimations iirc